artiq/coredevice/mirny.py

```diff
@@ -140,3 +151,143 @@ def write_ext(self, addr, length, data):
         if length < 32:
             data <<= 32 - length
         self.bus.write(data)
+
+
+class Almazny:
+    """
+    Almazny (High frequency mezzanine board for Mirny)
+
+    :param host_mirny - Mirny device Almazny is connected to
+    """
+
+    def __init__(self, dmgr, host_mirny):
+        self.mirny = dmgr.get(host_mirny)
+        self.att_mu = [0x3f] * 4
+        self.rf_switches = [0] * 4
+
+    @kernel
+    def init(self):
+        self._send_mezz_data(ALMAZNY_REG_CLEAR)  # reset the rest
+
+    @kernel
+    def att_to_mu(self, att):
+        """
+        Convert an attenuator setting in dB to machine units.
+
+        :param att: attenuator setting in dB [0-31.5]
+        :return: attenuator setting in machine units
+        """
+        if att > 31.5 or att < 0:
+            raise ValueError("Invalid Almazny attenuator settings!")
```

Comment:
You may want to only check after ``round`` - floating point computations are always imprecise and small rounding errors from the user code may otherwise unnecessarily trigger the error.